Unit test generated by RoostGPT

shivansh0017 opened the pull request:
```
Please pull these awesome changes in!
Test Name        - Java-repo-test
AI Type          - Open AI
AI Model         - gpt-4
Test Type        - Unit test
Use Type         - docker_ui
Language         - java
Test Framework   - Junit5
Packages to scan - com.thealgorithms.greedyalgorithms
```
[RoostGPT Logs](https://dev.roost.ai/roostgpt/logs?trigger_id=d450664f-592b-4fa3-aba8-9cc68eea39f2)

shivansh0017 committed:
Unit test generated by RoostGPT

Using AI Model gpt-4

shivansh0017 committed:
added missing import statements

shivansh0017 committed:
fixed syntax errors

shivansh0017 commented:
![image_720](https://github.com/shivansh0017/Java/assets/115993253/6a40aad4-3115-4f59-8c46-834efcf72140)

shivansh0017 commented:
![image](https://github.com/shivansh0017/Java/assets/115993253/03d2a172-3af2-4df3-98f6-0ec933db8168)

shivansh0017 commented:
[INFO] Results:
[INFO]
[ERROR] Failures: 
[ERROR]   FractionalKnapsackFractionalKnapsackTest.testFractionalKnapsackWithDifferentInputs:71 expected: <220> but was: <260>
[ERROR]   JobSequencingCompareToTest.compareHigherProfit:143 expected: <true> but was: <false>
[ERROR]   JobSequencingCompareToTest.compareLowerProfit:153 expected: <true> but was: <false>
[INFO]
[ERROR] Tests run: 33, Failures: 3, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  10.981 s
[INFO] Finished at: 2024-06-13T10:56:12+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.2.5:test (default-test) on project Java: There are test failures.
[ERROR]
[ERROR] Please refer to C:\Users\Shivansh\Desktop\Internship\Java-repo\Java\target\surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException

`No coverage report found after compilation errors were resolved for OpenAI Model`

harishagrawal commented on src/test/java/com/thealgorithms/greedyalgorithms/ActivitySelectionActivitySelectionTest.java:
```diff
@@ -0,0 +1,63 @@
+// <properties>
+//     <maven.compiler.source>1.8</maven.compiler.source>
+//     <maven.compiler.target>1.8</maven.compiler.target>
+// </properties>
```
```suggestion
// </properties>

package com.thealgorithms.greedyalgorithms;

import org.junit.jupiter.api.Test;
import java.util.ArrayList;
import java.util.Arrays;
import static org.junit.jupiter.api.Assertions.*;

public class ActivitySelectionActivitySelectionTest {

    @Test
    public void testActivitySelectionWithValidArrays() {
        int[] startTimes = {1, 3, 0, 5, 8, 5};
        int[] endTimes = {2, 4, 6, 7, 9, 9};
        ArrayList<Integer> actualResult = ActivitySelection.activitySelection(startTimes, endTimes);
        ArrayList<Integer> expectedResult = new ArrayList<>(Arrays.asList(0, 1, 3, 4));
        assertEquals(expectedResult, actualResult);
    }

    @Test
    public void testActivitySelectionWithOverlappingActivities() {
        int[] startTimes = {1, 3, 0, 5, 8, 5};
        int[] endTimes = {2, 4, 6, 9, 10, 9};
        ArrayList<Integer> actualResult = ActivitySelection.activitySelection(startTimes, endTimes);
        ArrayList<Integer> expectedResult = new ArrayList<>(Arrays.asList(0, 1, 4));
        assertEquals(expectedResult, actualResult);
    }

    @Test
    public void testActivitySelectionWithEmptyArrays() {
        int[] startTimes = {};
        int[] endTimes = {};
        ArrayList<Integer> actualResult = ActivitySelection.activitySelection(startTimes, endTimes);
        ArrayList<Integer> expectedResult = new ArrayList<>();
        assertEquals(expectedResult, actualResult);
    }

    @Test
    public void testActivitySelectionWithNullArrays() {
        int[] startTimes = null;
        int[] endTimes = null;
        assertThrows(NullPointerException.class, () -> ActivitySelection.activitySelection(startTimes, endTimes));
    }
}
```

shivansh0017 committed:
Improved Unit test generated by RoostGPT

Using AI Model gpt-4, for user feedback: Replace this line Job job = Job.of(jobName, processingTime, deadline); with MinimizingLateness.Job job = MinimizingLateness.Job.of(jobName, processingTime, deadline);
